test/libdnf/rpm/test_package.cpp

```diff
@@ -0,0 +1,346 @@
+    const auto reqs = get_pkg("filesystem-3.9-2.fc29.x86_64").get_requires();
+    const std::vector<std::string> expected = {"setup"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(reqs));
+}
+
+
+void RpmPackageTest::test_get_requires_pre() {
+    const auto prereqs = get_pkg("filesystem-3.9-2.fc29.x86_64").get_requires_pre();
+    // TODO no prereqs in test packages
+    const std::vector<std::string> expected = {};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(prereqs));
+}
+
+
+void RpmPackageTest::test_get_conflicts() {
+    const auto conflicts = get_pkg("nodejs-1:5.12.1-1.fc29.x86_64").get_conflicts();
+    const std::vector<std::string> expected = {"node <= 0.3.2-12"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(conflicts));
+}
+
+
+void RpmPackageTest::test_get_obsoletes() {
+    const auto obsoletes = get_pkg("npm-1:5.12.1-1.fc29.x86_64").get_obsoletes();
+    const std::vector<std::string> expected = {"npm < 3.5.4-6"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(obsoletes));
+}
+
+
+void RpmPackageTest::test_get_prereq_ignoreinst() {
+    const auto ipr = get_pkg("npm-1:5.12.1-1.fc29.x86_64").get_prereq_ignoreinst();
+    // TODO no prereq_ignoreinst (called Ipr: in libsolv testcases) in test packages
+    const std::vector<std::string> expected = {};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(ipr));
+}
+
+
+void RpmPackageTest::test_get_regular_requires() {
+    const auto reg_reqs = get_pkg("filesystem-3.9-2.fc29.x86_64").get_regular_requires();
+    // TODO doesn't test a difference between requires() and this, since there are no prereqs
+    const std::vector<std::string> expected = {"setup"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(reg_reqs));
+}
+
+
+void RpmPackageTest::test_get_recommends() {
+    const auto recommends = get_pkg("abcde-2.9.2-1.fc29.noarch").get_recommends();
+    const std::vector<std::string> expected = {"flac"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(recommends));
+}
+
+
+void RpmPackageTest::test_get_suggests() {
+    const auto suggests = get_pkg("abcde-2.9.2-1.fc29.noarch").get_suggests();
+    const std::vector<std::string> expected = {"lame"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(suggests));
+}
+
+
+void RpmPackageTest::test_get_enhances() {
+    const auto enhances = get_pkg("abcde-2.9.2-1.fc29.noarch").get_enhances();
+    // TODO no enhances in test packages
+    const std::vector<std::string> expected = {};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(enhances));
+}
+
+
+void RpmPackageTest::test_get_supplements() {
+    const auto supplements = get_pkg("glibc-langpack-af-2.28-9.fc29.x86_64").get_supplements();
+    const std::vector<std::string> expected = {"(glibc and (langpacks-af or langpacks-af_ZA))"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(supplements));
+}
+
+
+void RpmPackageTest::test_get_baseurl() {
+    CPPUNIT_ASSERT_EQUAL(std::string(), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_baseurl());
```

Comment:
Please could you investigate how to not have baseurl empty? Then set an additional test that will compare it. The comparison with empty string is not a good approach. Also some features could be tested in CI_DNF_STACK where more options are available.

Reply:
Please what's the status?

Reply:
The test is correct, `Package::get_baseurl()` returns a non-empty string only if the `xml:base` attribute is present in the package's location tag in `primary.xml`: `<location xml:base="http://example.com" href="x86_64/foo-1.0-1.x86_64.rpm"/>`. This isn't the case for the testing data we have now and adding this attribute is done using `createrepo_c`: `createrepo_c --baseurl "http://example.com" ...`.

This would best be done after the testing data are refactored, right now it's the generated data that are stored in git.

Reply:
But you already added or updated test data for tests therefore it can finished completely.

Reply:
What do you mean? That I can add this data as well? Have a look at https://github.com/rpm-software-management/libdnf/pull/1006/commits/8cc7aa4d21e9e149f58e82564c998d16caf11919. I've included the line I used to regenerate the repo data in the commit message. The value for the baseurl here needs to be part of that line.

Also, doing it this way, either all the packages have the `xml:base` set or they don't meaning I can't easily test both it being there and not.

I can add a simple script to git with the `createrepo_c` line so that it's at least part of git (come think of it I should do it anyway), but it's still just putting patches over patches on the already insufficient solution.

Reply:
Or you can add another repository for the test, just small one. As I already mentioned in this PR, the support of multiple repositories in test is useful.